usageReports/reservedInstances/get_ec2_data.go

```diff
@@ -0,0 +1,122 @@
+				"index": index,
+				"error": err.Error(),
+			})
+			return nil, http.StatusOK, errors.GetErrorMessage(ctx, err)
+		} else if cast, ok := err.(*elastic.Error); ok && cast.Details.Type == "search_phase_execution_exception" {
+			l.Error("Error while getting data from ES", map[string]interface{}{
+				"type":  fmt.Sprintf("%T", err),
+				"error": err,
+			})
+		} else {
+			l.Error("Query execution failed", map[string]interface{}{"error": err.Error()})
+		}
+		return nil, http.StatusInternalServerError, errors.GetErrorMessage(ctx, err)
+	}
+	return res, http.StatusOK, nil
+}
+
+// GetReservedInstancesMonthly does an elastic request and returns an array of reservations monthly report based on query params
+func GetReservedInstancesMonthly(ctx context.Context, params ReservedInstancesQueryParams) (int, []ReservationReport, error) {
+	res, returnCode, err := makeElasticSearchRequest(ctx, params, getElasticSearchReservedInstancesMonthlyParams)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	reservations, err := prepareResponseReservedInstancesMonthly(ctx, res)
+	if err != nil {
+		return http.StatusInternalServerError, nil, err
+	}
+	return http.StatusOK, reservations, nil
+}
+
+// GetReservedInstancesDaily does an elastic request and returns an array of daily report reservations based on query params
+func GetReservedInstancesDaily(ctx context.Context, params ReservedInstancesQueryParams, user users.User, tx *sql.Tx) (int, []ReservationReport, error) {
+	res, returnCode, err := makeElasticSearchRequest(ctx, params, getElasticSearchReservedInstancesDailyParams)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	accountsAndIndexes, returnCode, err := es.GetAccountsAndIndexes(params.AccountList, user, tx, es.IndexPrefixLineItems)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	params.AccountList = accountsAndIndexes.Accounts
+	params.IndexList = accountsAndIndexes.Indexes
+	costRes, _, _ := makeElasticSearchRequest(ctx, params, getElasticSearchCostParams)
+	reservations, err := prepareResponseReservedInstancesDaily(ctx, res, costRes)
+	if err != nil {
+		return http.StatusInternalServerError, nil, err
+	}
+	return http.StatusOK, reservations, nil
+}
+
+// GetReservedInstancesData gets Reserved Instances monthly reports based on query params, if there isn't a monthly report, it gets daily reports
+func GetReservedInstancesData(ctx context.Context, parsedParams ReservedInstancesQueryParams, user users.User, tx *sql.Tx) (int, []ReservationReport, error) {
+	accountsAndIndexes, returnCode, err := es.GetAccountsAndIndexes(parsedParams.AccountList, user, tx, reservedInstances.IndexPrefixReservedInstancesReport)
+	if err != nil {
+		return returnCode, nil, err
+	}
+	parsedParams.AccountList = accountsAndIndexes.Accounts
+	parsedParams.IndexList = accountsAndIndexes.Indexes
+	returnCode, monthlyReservations, err := GetReservedInstancesMonthly(ctx, parsedParams)
```

Comment:
make the code cleaner to remove every reference to monthly reports because you don't have them